### PR TITLE
Sync dashboard manifest with upstream files

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

This PR is targeted to sync downstream with upstream, it will need a follow up with the kfdefs for the correct overlay.
The main changes are:

- Remove old files
- Add a new rhods overlay with the changes for rhods

## Testing instructions

You can test the following kfdef:

```yaml
apiVersion: kfdef.apps.kubeflow.org/v1
kind: KfDef
metadata:
  name: rhods-dashboard
spec:
  applications:
  - kustomizeConfig:
      parameters:
      - name: odh-dashboard-image
        value: ${RELATED_IMAGE_ODH_DASHBOARD_IMAGE} //change this with the targeted image
      overlays:
        - rhods
      repoRef:
        name: manifests
        path: odh-dashboard
    name: odh-dashboard
  repos:
  - name: manifests
    uri: https://github.com/lucferbux/odh-manifests/tarball/sync-dashboardconfig-upstream
  version: v1.0.0
```

To check if the operator can add the overlay, the other testing method is:

1. Go to the `master` branch and run `kustomize build odh-dashboard/overlays/authentication`
2. Copy the text to a git differ
3. No switch to this branch
4. Run `kustomize build odh-dashboard/overlays/rhods`
5. Check the diff, there should be several additions and removals, most of them new changes that weren't tracked downstream, but all the names should be `rhods-dashboard` instead of `odh-dashboard`

Or install the live build:

`quay.io/lferrnan/rhods-operator-live-catalog:1.31.0-sync`

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message.
- [x] JIRA link(s): https://issues.redhat.com/browse/RHODS-10169, https://issues.redhat.com/browse/RHODS-10170
- [ ] The Jira story is acked.
- [x] Live build image: 
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work.
- [ ] QE contact acknowledges that this has been tested and is approved for merge.

### CODE REVIEW

**Comment by lucferbux** [comment]
@VaishnaviHire @zdtsw I'm creating this to sync upstream with downstream files, I'm removing the `authentication` overlay.
It's still a work in progress, I need to add the actual override values but tested the customize output of both and so far looks great, only `+37` changes (mostly rhods naming).
Will continue with this tomorrow.

File: odh-dashboard/base/deployment.yaml [hunk 1]
```diff
@@ -22,12 +22,12 @@ spec:
...
                       operator: In
                       values:
-                        - rhods-dashboard
+                        - odh-dashboard
                 topologyKey: topology.kubernetes.io/zone
-      serviceAccount: rhods-dashboard
+      serviceAccount: odh-dashboard
       containers:
-      - name: rhods-dashboard
-        image: $(odh-dashboard-image)
```
Comment (VaishnaviHire) [inline odh-dashboard/base/deployment.yaml:30]: Won't this be required to get CPaas image?

Comment (lucferbux) [reply odh-dashboard/base/deployment.yaml:30]: ah damn, that's the overlay i was missing, one sec, that might be the thing i was trying to fix that i totally thought i added in the overlay!

File: odh-dashboard/base/deployment.yaml [hunk 2]
```diff
@@ -1,16 +1,16 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: rhods-dashboard
+  name: odh-dashboard
 spec:
-  replicas: 5
+  replicas: 2
```
Comment (mwaykole) [inline odh-dashboard/base/deployment.yaml:6]: hi @lucferbux any specific reason for reducing the replica set count ?

Comment (lucferbux) [reply odh-dashboard/base/deployment.yaml:6]: Yes, we are not changing the replicas, the overlay will keep it as 5, we are syncing that file with upstream, and in odh we deploy 2 replicas.

**Comment by lucferbux** [comment]
Just to add more context, this PR will be followed up by https://github.com/red-hat-data-services/odh-manifests/pull/452

File: odh-dashboard/base/deployment.yaml (hunk 2 above)
Comment (zdtsw) [reply odh-dashboard/base/deployment.yaml:6]: the "real" value for rhods is set in odh-dashboard/overlays/rhods/deployment.yaml

File: odh-dashboard/base/routes.yaml [hunk 3]
```diff
@@ -4,14 +4,14 @@ metadata:
...
-  name: rhods-dashboard
+  name: odh-dashboard
 spec:
   port:
-    targetPort: 8080
+    targetPort: 8443
   tls:
     insecureEdgeTerminationPolicy: Redirect
-    termination: edge
+    termination: Reencrypt
```
Comment (zdtsw) [inline odh-dashboard/base/routes.yaml:13]:
```suggestion
    termination: reencrypt
```

The accepted values are `edge`, `passthrough` and `reencrypt`. All other values are silently ignored.

Comment (lucferbux) [reply odh-dashboard/base/routes.yaml:13]: mmm let me change that in upstream too then!

File: odh-dashboard/base/role.yaml [hunk 4]
```diff
@@ -101,24 +101,36 @@ rules:
...
-      - console.openshift.io
-    verbs:
+  - verbs:
       - get
       - list
+    apiGroups:
+      - console.openshift.io
     resources:
       - odhquickstarts
+  - apiGroups:
```
[the remark this answers is not in the document]
Comment (lucferbux) [reply odh-dashboard/base/role.yaml:125]: i copied latest upstream base folder, it should be the same as upstream, let me check again

File: odh-dashboard/base/deployment.yaml [hunk 5]
```diff
@@ -56,5 +56,72 @@ spec:
           periodSeconds: 30
           successThreshold: 1
           failureThreshold: 3
-      imagePullSecrets:
```
[the remark this answers is not in the document]
Comment (lucferbux) [reply odh-dashboard/base/deployment.yaml:59]: done

File: odh-dashboard/base/routes.yaml (hunk 3 above)
Comment (lucferbux) [reply odh-dashboard/base/routes.yaml]: done

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Comment by lucferbux** [comment]
Holding this until 1.33, since we want to have a clean 1.32 release for WatsonX

**Comment by lucferbux** [comment]
@andrewballantyne @VaishnaviHire @zdtsw I've rebased the branch and carry over latest changes from upstream (there were none) so I think it's ready to go.

**Comment by lucferbux** [comment]
Ok, I've run again the tests and seems that everything is ok, can we proceed with this?
@VaishnaviHire @zdtsw can you take a look again, I'm sorry, it's just for double checking, I want to push this forward asap, to not let this slip another sprint

**Comment by zdtsw** [comment]
> Ok, I've run again the tests and seems that everything is ok, can we proceed with this? @VaishnaviHire @zdtsw can you take a look again, I'm sorry, it's just for double checking, I want to push this forward asap, to not let this slip another sprint

me too :D

**Review by zdtsw — APPROVED** [review]
/lgtm

**Pull request merged**